cirq/study/trial_result.py

```diff
@@ -271,3 +271,21 @@ def __eq__(self, other):
         if not isinstance(other, type(self)):
             return NotImplemented
         return self.data.equals(other.data) and self.params == other.params
+
+    def _measurement_shape(self):
+        return self.params, {k: v.shape[1] for k, v in self.measurement.items()}
+
+    def __add__(self, other: 'cirq.TrialResult') -> 'cirq.TrialResult':
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        if self._measurement_shape() != other._measurement_shape():
```

Comment:
I think you've got the comparison backwards here.

Keep it like it is, but move the exception case to be conditional instead of the correct case. (Make it a guard statement.)

Reply:
Oops, I found this comment too late, is it OK the way I have it now or would you prefer me to revert back ?